【请求帮助】【win11 Arm 24H2】Process Uncaught Exception: Error: The specified module could not be found.

<!-- 如果搜索过但未找到，请将 `[ ]` 替换为 `[x]` -->

- [ ] 你是否在现有 [Issue列表](/docmirror/dev-sidecar/issues) 中搜索过相同问题，但未找到？

### Ⅰ. 请说明操作系统及DS的版本号：

- 操作系统：mac下的虚拟机 `win11 Arm 24H2`
- DS版本号：[2.0.0-beta-2: Try support arm64](https://github.com/docmirror/dev-sidecar/releases/tag/test-arm64-2) 版本

### Ⅱ. 问题描述：

安装后，点击图标没反应 进程在运行，有报错：

```log
[2024-11-27T10:58:35.422] [ERROR] core - Process Uncaught Exception: Error: The specified module could not be found.
\\?\C:\Users\yeyongqi\AppData\Local\Temp\2a66ed48-87a6-448e-a388-cd3740fe4650.tmp.node
    at process.func [as dlopen] (node:electron/js2c/asar_bundle:5:1812)
    at Module._extensions..node (node:internal/modules/cjs/loader:1203:18)
    at Object.func [as .node] (node:electron/js2c/asar_bundle:5:2039)
    at Module.load (node:internal/modules/cjs/loader:988:32)
    at Module._load (node:internal/modules/cjs/loader:829:12)
    at c._load (node:electron/js2c/asar_bundle:5:13343)
    at Module.require (node:internal/modules/cjs/loader:1012:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (C:\Program Files (x86)\dev-sidecar\resources\app.asar\node_modules\@natmri\platform-napi\index.js:94:29)
    at Module._compile (node:internal/modules/cjs/loader:1118:14) {
  code: 'ERR_DLOPEN_FAILED'
}
```

估计是操作系统缺少DLL库导致，尝试通过调整依赖 `@natmri/platform-napi` 的源码来解决掉这个问题，但还是无法得到解决。
目前暂时捕获了以上异常，让DS能够正常启动并使用，但 `windows系统下重启后因系统代理未关闭导致网络不可用` 的问题又会出现。

### Ⅲ. 期望的结果：

1. 捕获异常的代码去掉
2. 不使用DLL修复工具修复DLL
4. 能正常运行DS：代理服务、系统代理都能正常启动

--- a/.github/workflows/build-and-release.yml
+++ b/.github/workflows/build-and-release.yml
@@ -95,9 +95,9 @@ jobs:
           dir || ls -lah;
 
           echo "======================================================================";
-          echo "pnpm install --force";
+          echo "pnpm install";
           echo "--------------------";
-          pnpm install --force;
+          pnpm install;
 
       - name: 'npm run electron:build'
         run: |

--- a/.github/workflows/test-and-upload.yml
+++ b/.github/workflows/test-and-upload.yml
@@ -98,9 +98,9 @@ jobs:
           dir || ls -lah;
 
           echo "======================================================================";
-          echo "pnpm install --force";
+          echo "pnpm install";
           echo "--------------------";
-          pnpm install --force;
+          pnpm install;
 
       - name: 'npm run electron:build'
         run: |

--- a/package.json
+++ b/package.json
@@ -13,5 +13,11 @@
     "@antfu/eslint-config": "^3.9.1",
     "eslint": "^9.15.0",
     "eslint-plugin-format": "^0.1.2"
+  },
+  "pnpm": {
+    "supportedArchitectures": {
+      "os": ["current"],
+      "cpu": ["x64", "arm64", "ia32"]
+    }
   }
 }

--- a/packages/core/package.json
+++ b/packages/core/package.json
@@ -16,7 +16,7 @@
     "test": "mocha"
   },
   "dependencies": {
-    "@starknt/sysproxy": "^0.0.1",
+    "@starknt/sysproxy": "^0.0.2",
     "@vscode/sudo-prompt": "^9.3.1",
     "fix-path": "^3.0.0",
     "iconv-lite": "^0.6.3",

--- a/packages/gui/package.json
+++ b/packages/gui/package.json
@@ -22,8 +22,8 @@
   "dependencies": {
     "@docmirror/dev-sidecar": "workspace:*",
     "@docmirror/mitmproxy": "workspace:*",
-    "@starknt/shutdown-handler-napi": "^0.0.2",
-    "@starknt/sysproxy": "^0.0.1",
+    "@starknt/shutdown-handler-napi": "^0.0.3",
+    "@starknt/sysproxy": "^0.0.2",
     "@vscode/sudo-prompt": "^9.3.1",
     "adm-zip": "^0.5.16",
     "ant-design-vue": "^1.7.8",

